packages/client/src/editor/components/Tools.tsx

```diff
@@ -1,17 +1,37 @@
-import React from 'react';
+import React, { useRef } from 'react';
 import { ToggleButton, ToggleButtonGroup } from '@material-ui/lab';
 import { useRecoilState } from 'recoil';
-import { modeState } from '../state';
+import { modeState, imgState } from '../state';
 import { Mode } from '../../types';
 import NoteAddIcon from '@material-ui/icons/NoteAdd';
+import ImageIcon from '@material-ui/icons/Image';
 import Draw from '../../cursors/Draw';
 import Erase from '../../cursors/Erase';
 
 export default function Tools() {
   const [mode, setMode] = useRecoilState(modeState);
+  const [, setImgData] = useRecoilState(imgState);
+  const imageUpload = useRef(null);
 
   const handleMode = (event: React.MouseEvent<HTMLElement>, newMode: Mode) => {
-    setMode(newMode);
+    if (newMode === 'image') {
+      // @ts-ignore
```

Comment:
Noting here that this is because `imageUpload` is typed as optinally `null`, but it gets set below by `input`.